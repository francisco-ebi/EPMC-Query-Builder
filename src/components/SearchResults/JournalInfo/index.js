const formatDate = date => {
  return new Date(date).toLocaleDateString(
    'en-GB',
    {day: "2-digit", month: "short", year: "numeric"}
  )
}

export default function JournalInfo({
  publication: {
    journalInfo,
    pageInfo,
    firstPublicationDate,
    bookOrReportDetails
  }
}) {

  if (journalInfo) {
    return (
      <p className="journal-info">
        {journalInfo.journal.isoabbreviation}
        {journalInfo.volume ? `, ${journalInfo.volume}` : null}
        {journalInfo.issue ? `(${journalInfo.issue})` : null}
        {pageInfo ? `:${pageInfo}` : null}
        , {formatDate(firstPublicationDate)}
      </p>
    );
  } else {
    return (
      <p className="publisher-info">
        {bookOrReportDetails.publisher},
        {bookOrReportDetails.yearOfPublication}
      </p>
    );
  }

}