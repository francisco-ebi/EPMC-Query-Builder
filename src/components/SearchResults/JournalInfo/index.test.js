import renderer from 'react-test-renderer';
import JournalInfo from './index';

it('should output journal info if publication has data', () => {
  const publication = {
    journalInfo: {
      issue: "9",
      volume: "24",
      journalIssueId: 3412000,
      dateOfPublication: "2022 Sep",
      monthOfPublication: 9,
      yearOfPublication: 2022,
      printPublicationDate: "2022-09-01",
      journal: {
          title: "Journal of feline medicine and surgery",
          medlineAbbreviation: "J Feline Med Surg",
          essn: "1532-2750",
          issn: "1098-612X",
          isoabbreviation: "J Feline Med Surg",
          nlmid: "100897329"
      }
    },
    pageInfo: "847-852",
    firstPublicationDate: "2022-09-01"
  };
  const component = renderer.create(
    <JournalInfo publication={publication}/>
  );
  const testInstance = component.root;
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(
    testInstance.findByProps({ className: "journal-info" })
    .children
    .join('')
  ).toEqual('J Feline Med Surg, 24(9):847-852, 01 Sept 2022');
});

it('should output bookOrReport details if does not have journal info', () => {
  const publication = {
    bookOrReportDetails: {
      publisher: "bioRxiv",
      yearOfPublication: 2022
    }
  };
  const component = renderer.create(
    <JournalInfo publication={publication}/>
  );
  const testInstance = component.root;
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(
    testInstance.findByProps({ className: "publisher-info" })
    .children
    .join('')
  ).toBe('bioRxiv,2022');
})