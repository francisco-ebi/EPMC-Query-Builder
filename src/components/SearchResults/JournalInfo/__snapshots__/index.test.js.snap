// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should output bookOrReport details if does not have journal info 1`] = `
<p
  className="publisher-info"
>
  bioRxiv
  ,
  2022
</p>
`;

exports[`should output journal info if publication has data 1`] = `
<p
  className="journal-info"
>
  J Feline Med Surg
  , 24
  (9)
  :847-852
  , 
  01 Sept 2022
</p>
`;
